### ISSUE TITLE
Space on both sides of operator method definition

### ISSUE THREAD

**Issue by durban** [scalameta/scalafmt#915]
Feature request: an option to surround symbolic method names with exactly one space on both sides.

- **Version**: 0.6.8
- **Integration**: sbt
- **Configuration**: default

## Problem

Scalafmt formats code like this:
```scala
trait Test[A] {
  def <=>[B](that: Test[B]): Int
}
```

## Expectation

I would like the formatted output to look like this:
```scala
trait Test[A] {
  def <=> [B](that: Test[B]): Int
}
```

Note the space immediately after the `<=>` method name. It mirrors how it would typically appear at callsites: `x <=> y`. (Obviously the `<=>` operator is just an example.)

**Comment by olafurpg** [comment]
Thank you for reporting! This should be possible to accommodate. What do you think would be a good name for this option? I propose `spaces.aroundSymbolicOperatorDefinitions`.

The tutorial https://github.com/scalameta/scalafmt/blob/master/Tutorial.md should contain all the necessary information to find the place to implement this and test it.

**Comment by ShaneDelmore** [comment]
Doesn't this happen on application, not just definition?  I seem to recall seeing it in scalatest with foo should ===(bar) but can't recall if it has been fixed or not.  If so, then I like a name closer to the original reported issue,  spaces.aroundSymbolicMethodNames to signify the space should be at both the definition site and the usage site.

**Comment by olafurpg** [comment]
@ShaneDelmore You may be referring  to the `spaces.afterTripleEquals = true` setting to get this output`a should be === (x)`. I'm personally very much against that setting because it makes the illusion that `===(x)` is an infix operator when in fact it's a regular Term.Apply. Infix operators always have space at call-site, so it's not necessary to have a setting for that. 

However, now that I think about it maybe `spaces.aroundSymbolOperatorDefnSite` is more consistent with other callSite/defnSite settings.

**Comment by durban** [comment]
I think `spaces.aroundSymbolOperatorDefnSite` is okay. Maybe even `spaces.aroundOperatorDefnSite`. But english is not my first language, so you shouldn't believe me about naming things :-)

**Comment by olafurpg** [comment]
Fixed in #1188 by @slavaschmidt  🎉
